src/main/java/com/google/cloud/resourcemanager/ConstraintInfo.java

```diff
@@ -0,0 +1,315 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import com.google.api.services.cloudresourcemanager.model.BooleanConstraint;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+import java.util.Objects;
+
+/** A Google Cloud Resource Manager constraint metadata object. */
+public class ConstraintInfo implements Serializable {
+
+  private static final long serialVersionUID = 9148970963697734236L;
+
+  static final Function<
+          com.google.api.services.cloudresourcemanager.model.Constraint, ConstraintInfo>
+      FROM_PB_FUNCTION =
+          new Function<
+              com.google.api.services.cloudresourcemanager.model.Constraint, ConstraintInfo>() {
+            @Override
+            public ConstraintInfo apply(
+                com.google.api.services.cloudresourcemanager.model.Constraint pb) {
+              return ConstraintInfo.fromPb(pb);
+            }
+          };
+  static final Function<
+          ConstraintInfo, com.google.api.services.cloudresourcemanager.model.Constraint>
+      TO_PB_FUNCTION =
+          new Function<
+              ConstraintInfo, com.google.api.services.cloudresourcemanager.model.Constraint>() {
+            @Override
+            public com.google.api.services.cloudresourcemanager.model.Constraint apply(
+                ConstraintInfo constraintInfo) {
+              return constraintInfo.toPb();
+            }
+          };
+
+  private BooleanConstraint booleanConstraint;
+  private String constraintDefault;
+  private String description;
+  private String displayName;
+  private ListConstraint listConstraint;
+  private String name;
+  private Integer version;
+
+  /**
+   * A Constraint that allows or disallows a list of string values, which are configured by an
+   * Organization's policy administrator with a Policy.
+   */
+  public static class ListConstraint implements Serializable {
+
+    private static final long serialVersionUID = -2133042982786959352L;
+
+    private final String suggestedValue;
+    private final Boolean supportsUnder;
+
+    ListConstraint(String suggestedValue, Boolean supportsUnder) {
+      this.suggestedValue = suggestedValue;
+      this.supportsUnder = supportsUnder;
+    }
+
+    public String getSuggestedValue() {
+      return suggestedValue;
+    }
+
+    public Boolean getSupportsUnder() {
+      return supportsUnder;
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("suggestedValue", getSuggestedValue())
+          .add("supportsUnder", getSupportsUnder())
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(getSuggestedValue(), getSupportsUnder());
+    }
+
+    @Override
+    public final boolean equals(Object obj) {
```

Comment:
Weird case but if equals is final, hashCode should be too, and I think the class should be as well.

Reply:
done